pkg/network/bridge/interfacecleaner.go

```diff
@@ -0,0 +1,66 @@
+package bridge
+
+import (
+	"sync"
+	"strconv"
+	
+	"github.com/fsnotify/fsnotify"
+	"github.com/vishvananda/netlink"
+)
+
+type interfaceCleaner struct {
+	hostInterfaceByPID map[int]string
+	mutex              *sync.Mutex
+}
+
+func newInterfaceCleaner() *interfaceCleaner {
+	return &interfaceCleaner{
+		hostInterfaceByPID: make(map[int]string),
+		mutex:              new(sync.Mutex),
+	}
+}
+
+func (ic *interfaceCleaner) Run() {
+	watcher, err := fsnotify.NewWatcher()
+	if err != nil {
+		panic(err)
+	}
+	defer watcher.Close()
+
+	err = watcher.Add("/proc")
+	if err != nil {
+		panic(err)
+	}
+
+	for {
+		select {
+		case event := <-watcher.Events:
+			if event.Op&fsnotify.Remove == fsnotify.Remove {
+				ic.mutex.Lock()
+
+				pid, err := strconv.Atoi(event.Name)
+				if err != nil {
+					panic(err)
+				}
+
+				peer, err := netlink.LinkByName(ic.hostInterfaceByPID[pid])
```

Comment:
probably better to check if the pid exists in the map. if its not there, we can probably just ignore. it could be that some other process ended, unrelated to the launcher?

Reply:
also, if you cannot get the link, IMO, don't panic, just log warning and continue - since this is a cleanup operation, its ok if the resource is already cleaned up

Reply:
Done